libs/common-utils/src/environments.ts

```diff
@@ -1,7 +1,21 @@
 import { registerOnWindow } from './misc'
 
-const getRegex = (regex: string | undefined) => (regex ? new RegExp(regex, 'i') : undefined)
+const DEFAULT_ENVIRONMENTS_REGEX: Record<Envs, string> = {
+  LOCAL: '^(:?localhost:d{2,5}|(?:127|192)(?:.[0-9]{1,3}){3})',
+  PR: '^prd+--cowswap.review|^(swap-dev-git-[wd-]+|swap-w{9}-)cowswap.vercel.app',
+  DEV: '^(dev.swap.cow.fi|swap-develop.vercel.app)',
+  STAGING: '^(staging.swap.cow.fi|swap-staging.vercel.app)',
+  PROD: '^(swap.cow.fi|swap-prod.vercel.app)$',
+  BARN: '^(barn.cow.fi|swap-barn.vercel.app)$',
+  ENS: '(:?^cowswap.eth|ipfs)',
+}
+
+function getRegex(env: Envs) {
+  const regex = process.env[`REACT_APP_DOMAIN_REGEX_${env}`] || DEFAULT_ENVIRONMENTS_REGEX[env]
```

Comment:
Nitpick:
I think we even don't need regexes to be stored in ENV variables. Because they are static and unlikely will be changed

Reply:
It's just for other UIs and forks

Reply:
Also, gives us flexibilities to deploy in other domains and only change the envs